evalml/pipelines/utils.py

```diff
@@ -65,6 +66,14 @@ def _get_preprocessing_components(
     """
 
     pp_components = []
+
+    if problem_type in [ProblemTypes.REGRESSION, ProblemTypes.TIME_SERIES_REGRESSION]:
+        for each_action in TargetDistributionDataCheck(problem_type).validate(X, y)[
```

Comment:
Gotcha, this makes sense. Personally, I'm not sure how much I like doing this check, since it seems we need to iterate through the datacheck to figure out if we need to add this component, but I can't think of another way to do it. Looks fine for now though!

Reply:
Yeah the reason it's set up like this is if we add more target checks in the future